openpype/plugins/publish/integrate_new.py

```diff
@@ -716,6 +710,62 @@ def get_subset(self, asset, instance):
 
         return subset
 
+    def _set_subset_group(self, instance, subset_id):
+        """
+            Mark subset as belonging to group in DB.
+
+            Uses Settings > Global > Publish plugins > IntegrateAssetNew
+
+            Args:
+                instance (dict): processed instance
+                subset_id (str): DB's subset _id
+
+        """
+        # add group if available
+        integrate_new_sett = (instance.context.data["project_settings"]
+                                                   ["global"]
+                                                   ["publish"]
+                                                   ["IntegrateAssetNew"])
+
+        profiles = integrate_new_sett["subset_grouping_profiles"]
+
+        filtering_criteria = {
+            "families": instance.data["anatomyData"]["family"],
+            "hosts": instance.data["anatomyData"]["app"],
+            "tasks": instance.data["anatomyData"]["task"] or
+                io.Session["AVALON_TASK"]
+        }
+        matching_profile = filter_profiles(profiles, filtering_criteria)
+
+        filled_template = None
+        fill_pairs = None
+        if matching_profile:
+            template = matching_profile["template"]
+            fill_pairs = {
+                "family": filtering_criteria["families"],
+                "task": filtering_criteria["tasks"],
+                "Family": filtering_criteria["families"].capitalize(),
+                "Task": filtering_criteria["tasks"].capitalize()
```

Comment:
maybe we could do this a bit less explicit, so it would look at the input and match the capitalisation of the value? I believe @iLLiCiTiT was already doing this same for creator somewhere.